datafusion/core/src/datasource/datasource.rs

```diff
@@ -77,6 +78,11 @@ pub trait TableProvider: Sync + Send {
     ) -> Result<TableProviderFilterPushDown> {
         Ok(TableProviderFilterPushDown::Unsupported)
     }
+
+    /// Get statistics for this table, if available
+    fn statistics(&self) -> Option<Statistics> {
```

Comment:
I wonder if any of the built in providers can give statistic (I am thinking Parquet) 🤔  

Definitely something for a follow on PR

Reply:
:+1: Filed as https://github.com/apache/arrow-datafusion/issues/3988